modules/apps/layout/layout-reports-web/src/main/java/com/liferay/layout/reports/web/internal/display/context/LayoutReportsDisplayContext.java

```diff
@@ -0,0 +1,189 @@
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.portlet.RenderRequest;
+
+/**
+ * @author Cristina González
+ */
+public class LayoutReportsDisplayContext {
+
+	public LayoutReportsDisplayContext(
+		GroupLocalService groupLocalService,
+		LayoutLocalService layoutLocalService,
+		LayoutSEOLinkManager layoutSEOLinkManager, Language language,
+		Portal portal, RenderRequest renderRequest) {
+
+		_groupLocalService = groupLocalService;
+		_layoutLocalService = layoutLocalService;
+		_layoutSEOLinkManager = layoutSEOLinkManager;
+		_language = language;
+		_portal = portal;
+		_renderRequest = renderRequest;
+
+		_themeDisplay = (ThemeDisplay)renderRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+	}
+
+	public Map<String, Object> getData() {
+		if (_data != null) {
+			return _data;
+		}
+
+		Layout layout = _layoutLocalService.fetchLayout(
+			_themeDisplay.getPlid());
+
+		_data = Collections.singletonMap(
+			"context",
+			HashMapBuilder.<String, Object>put(
+				"assetsPath",
+				_portal.getPathContext(_renderRequest) + "/assets/"
+			).put(
+				"canonicalURLs", _getCanonicalURLs(layout)
+			).put(
+				"defaultLanguageId",
+				LocaleUtil.toBCP47LanguageId(_getDefaultLocale(layout))
+			).put(
+				"showButton", false
+			).put(
+				"validConnection", false
+			).build());
+
+		return _data;
+	}
+
+	private Map<Locale, String> _getAlternateURLs(
+		String currentCompleteURL, Layout layout) {
+
+		try {
+			return _portal.getAlternateURLs(
+				currentCompleteURL, _themeDisplay, layout);
+		}
+		catch (PortalException portalException) {
+			_log.error(portalException, portalException);
+		}
+
+		return Collections.emptyMap();
+	}
+
+	private String _getCanonicalURL(String currentCompleteURL, Layout layout) {
+		try {
+			return _portal.getCanonicalURL(
+				currentCompleteURL, _themeDisplay, layout);
+		}
+		catch (PortalException portalException) {
+			_log.error(portalException, portalException);
+		}
+
+		return StringPool.BLANK;
+	}
+
+	private List<Map<String, Object>> _getCanonicalURLs(Layout layout) {
+		String currentCompleteURL = _portal.getCurrentCompleteURL(
+			_portal.getHttpServletRequest(_renderRequest));
+
+		String canonicalURL = _getCanonicalURL(currentCompleteURL, layout);
+
+		Map<Locale, String> alternateURLs = _getAlternateURLs(
+			currentCompleteURL, layout);
+
+		return Optional.ofNullable(
+			_groupLocalService.fetchGroup(layout.getGroupId())
+		).map(
+			Group::getGroupId
+		).map(
+			_language::getAvailableLocales
+		).orElseGet(
+			Collections::emptySet
+		).stream(
+		).map(
+			locale -> HashMapBuilder.<String, Object>put(
+				"canonicalURL",
+				() -> {
+					LayoutSEOLink layoutSEOLink =
+						_layoutSEOLinkManager.getCanonicalLayoutSEOLink(
+							layout, locale, canonicalURL, alternateURLs);
+
+					return layoutSEOLink.getHref();
+				}
+			).put(
+				"languageId", LocaleUtil.toBCP47LanguageId(locale)
+			).put(
+				"selected", false
```

Comment:
I suppose this will be used when changing the language in the dropdown. Are we going to do that action in this pull request? In the first render, the default language should have `selected: true`, right?

Reply:
No, selected is not used, this field should be deleted. With JS we can use the selected, as we do it now.

Reply:
Ok, I let you remove it then :)